### ISSUE TITLE
`ExistValidator` broken in case `$targetClass` is NOSQL ActiveRecord

### ISSUE THREAD

**Issue by klimov-paul** [yiisoft/yii2#14533]
### What steps will reproduce the problem?

- Create DB model with reference to MongoDB (or any NOSQL AR)
- Add validation on reference attribute with `targetClass` referencing MongoDB  model
- run DB model validation

```php
class DbModel
{
    public function rules()
    {
        return [
            ['mongodbReferenceId', 'exist', 'targetAttribute' => 'id', 'targetClass' => 'app\models\mongodb\MongoDbReference'],
            // ...
        ];
    }
}
```

### What is the expected result?

Validation runs

### What do you get instead?

yii\base\UnknownMethodException excpetion is thrown:
Calling unknown method: yii\mongodb\ActiveQuery::getTablesUsedInFrom()

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.13
| PHP version      | 5.6
| Operating system | CentOS

**Comment by klimov-paul** [comment]
Bug has been introduced at commit 99a02892210d6b3c03a4e6e036af5b30cae681d4
The reason is check `!$model instanceof ActiveRecord` make no sense, because `$this->getTargetClass($model)` may return different class name, which may be not `yii\db\ActiveRecord`.

**Comment by klimov-paul** [comment]
Same appear for `UniqueValidator`

**Comment by klimov-paul** [comment]
Resolved by commit 2466d397ae0cefbc048be0586487527037612e66
